### PR TITLE
Create an endpoint to get client's information

### PR DESCRIPTION
**Pull request by Anastasia-Troian**

### CODE REVIEW

File: Backend/Core/DTO/UserDTO/UserProfileInfoDTO.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Core.DTO.UserDTO
```
Comment (marynakernychna) [inline Backend/Core/DTO/UserDTO/UserProfileInfoDTO.cs]: Rename the folder to "User" & namespace to"Core.DTO.User"

File: Backend/Core/Helpers/ApplicationProfiles/UserProfiles.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using AutoMapper;
+using Core.DTO.UserDTO;
+using Core.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Core.Helpers.ApplicationProfiles
```
Comment (marynakernychna) [inline Backend/Core/Helpers/ApplicationProfiles/UserProfiles.cs]: Rename the folder to "User"

File: Backend/Core/Helpers/ApplicationProfiles/UserProfiles.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿using AutoMapper;
+using Core.DTO.UserDTO;
+using Core.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (marynakernychna) [inline Backend/Core/Helpers/ApplicationProfiles/UserProfiles.cs]: Remove the extra usings

File: Backend/Core/Helpers/ApplicationProfiles/UserProfiles.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Core.Helpers.ApplicationProfiles
+{
+    public class UserProfiles : Profile
+    {
+        public UserProfiles()
+        {
+            CreateMap<User, UserProfileInfoDTO>().ReverseMap();
```
Comment (marynakernychna) [inline Backend/Core/Helpers/ApplicationProfiles/UserProfiles.cs]: We don't need the reverse map now

File: Backend/Degree project/Controllers/UserController.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        private readonly IUserService _userService;
+
+        public UsersController(
+            IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        [HttpGet("user-info")]
+        public async Task<ActionResult> GetUserInfo()
```
Comment (marynakernychna) [inline Backend/Degree project/Controllers/UserController.cs]: Use IActionResult instead of ActionResult

Comment (marynakernychna) [inline Backend/Degree project/Controllers/UserController.cs]: Add "Async" to the end of the method name

File: Backend/Degree project/Controllers/UserController.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        {
+            var userId = _userService.GetCurrentUserNameIdentifier(User);
+            var userInfo = await _userService.GetUserProfileInfoAsync(userId);
+            return Ok(userInfo);
```
Comment (marynakernychna) [inline Backend/Degree project/Controllers/UserController.cs:27]: Add the extra line before return

**Review by marynakernychna — COMMENTED** [review]
Add validation

File: Backend/Core/Resources/ErrorMessages.Designer.cs [hunk 7]
```diff
@@ -77,5 +77,16 @@ internal class ErrorMessages {
                 return ResourceManager.GetString("InvalidCredentials", resourceCulture);
             }
         }
+
+        /// <summary>
+        ///   Looks up a localized string similar to User not found.
+        /// </summary>
+        internal static string UserNotFound
```
Comment (Grasmus) [inline Backend/Core/Resources/ErrorMessages.Designer.cs:84]: Leave the editing of this file to the system

File: Backend/Core/Helpers/ApplicationProfiles/User.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿using AutoMapper;
+using Core.DTO.User;
+
+namespace Core.Helpers.ApplicationProfiles
+{
+    public class User : Profile
```
Comment (Grasmus) [inline Backend/Core/Helpers/ApplicationProfiles/User.cs]: I think it`s better to name such classes as *RelatedEntityName*Profile. For instance, "UserProfile"

File: Backend/Core/Helpers/ApplicationProfiles/User.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+    {
+        public User()
+        {
+            CreateMap<Entities.User, UserProfileInfoDTO>();
```
Comment (Grasmus) [inline Backend/Core/Helpers/ApplicationProfiles/User.cs]: Do not create conflict names of two not related classes

**Review by Grasmus — APPROVED** [review]

**Pull request merged**